[BUG] Reported crash in FileDataStorageManager

Crash reported in Google console

```
java.lang.RuntimeException: 
  at android.os.AsyncTask$4.done (AsyncTask.java:399)
  at java.util.concurrent.FutureTask.finishCompletion (FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException (FutureTask.java:252)
  at java.util.concurrent.FutureTask.run (FutureTask.java:271)
  at android.os.AsyncTask$SerialExecutor$1.run (AsyncTask.java:289)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at java.lang.Thread.run (Thread.java:919)
Caused by: java.lang.NullPointerException: 
  at com.owncloud.android.datamodel.FileDataStorageManager.<init> (Unknown Source:7)
  at com.owncloud.android.files.services.AvailableOfflineSyncJobService$AvailableOfflineJobTask.doInBackground (AvailableOfflineSyncJobService.java:81)
  at com.owncloud.android.files.services.AvailableOfflineSyncJobService$AvailableOfflineJobTask.doInBackground (AvailableOfflineSyncJobService.java:61)
  at android.os.AsyncTask$3.call (AsyncTask.java:378)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
```

Happening only in Android 10

--- a/owncloudApp/src/main/java/com/owncloud/android/files/services/AvailableOfflineHandler.java
+++ b/owncloudApp/src/main/java/com/owncloud/android/files/services/AvailableOfflineHandler.java
@@ -40,11 +40,9 @@ public class AvailableOfflineHandler {
     // configuration
     private static final int JOB_ID_AVAILABLE_OFFLINE = 2;
 
-    private String mAccountName;
     private JobScheduler mJobScheduler;
 
-    public AvailableOfflineHandler(Context context, String accountName) {
-        mAccountName = accountName;
+    public AvailableOfflineHandler(Context context) {
         mJobScheduler = (JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
     }
 
@@ -66,8 +64,6 @@ public class AvailableOfflineHandler {
         PersistableBundle extras = new PersistableBundle();
 
         extras.putInt(Extras.EXTRA_AVAILABLE_OFFLINE_SYNC_JOB_ID, JOB_ID_AVAILABLE_OFFLINE);
-
-        extras.putString(Extras.EXTRA_ACCOUNT_NAME, mAccountName);
 
         builder.setExtras(extras);
 

--- a/owncloudApp/src/main/java/com/owncloud/android/files/services/AvailableOfflineSyncJobService.java
+++ b/owncloudApp/src/main/java/com/owncloud/android/files/services/AvailableOfflineSyncJobService.java
@@ -33,7 +33,6 @@ import com.owncloud.android.datamodel.FileDataStorageManager;
 import com.owncloud.android.datamodel.OCFile;
 import com.owncloud.android.lib.common.operations.RemoteOperationResult;
 import com.owncloud.android.operations.SynchronizeFileOperation;
-import com.owncloud.android.utils.Extras;
 import com.owncloud.android.utils.FileStorageUtils;
 import timber.log.Timber;
 
@@ -70,9 +69,8 @@ public class AvailableOfflineSyncJobService extends JobService {
         @Override
         protected JobParameters doInBackground(JobParameters... jobParams) {
 
-            String accountName = jobParams[0].getExtras().getString(Extras.EXTRA_ACCOUNT_NAME);
-
-            Account account = AccountUtils.getOwnCloudAccountByName(mAvailableOfflineJobService, accountName);
+            Account account =
+                    AccountUtils.getCurrentOwnCloudAccount(mAvailableOfflineJobService.getApplicationContext());
 
             FileDataStorageManager fileDataStorageManager = new FileDataStorageManager(
                     mAvailableOfflineJobService, account, mAvailableOfflineJobService.getContentResolver()
@@ -131,6 +129,10 @@ public class AvailableOfflineSyncJobService extends JobService {
             }
 
             Account account = AccountUtils.getOwnCloudAccountByName(mAvailableOfflineJobService, accountName);
+            if (account == null) {
+                Timber.w("Account '" + accountName + "' not found in account manager. Aborting Sync operation...");
+                return;
+            }
 
             FileDataStorageManager storageManager =
                     new FileDataStorageManager(

--- a/owncloudApp/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java
+++ b/owncloudApp/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java
@@ -32,13 +32,11 @@ import android.webkit.MimeTypeMap;
 
 import androidx.fragment.app.DialogFragment;
 import com.owncloud.android.R;
-import com.owncloud.android.authentication.AccountUtils;
 import com.owncloud.android.datamodel.OCFile;
 import com.owncloud.android.domain.sharing.shares.model.OCShare;
 import com.owncloud.android.files.services.AvailableOfflineHandler;
 import com.owncloud.android.files.services.FileDownloader.FileDownloaderBinder;
 import com.owncloud.android.files.services.FileUploader.FileUploaderBinder;
-import com.owncloud.android.lib.resources.status.OwnCloudVersion;
 import com.owncloud.android.presentation.ui.sharing.ShareActivity;
 import com.owncloud.android.services.OperationsService;
 import com.owncloud.android.ui.activity.FileActivity;
@@ -257,8 +255,7 @@ public class FileOperationsHelper {
 
             if (success) {
                 // Schedule job to check to watch for local changes in available offline files and sync them
-                AvailableOfflineHandler availableOfflineHandler =
-                        new AvailableOfflineHandler(mFileActivity, mFileActivity.getAccount().name);
+                AvailableOfflineHandler availableOfflineHandler = new AvailableOfflineHandler(mFileActivity);
                 availableOfflineHandler.scheduleAvailableOfflineJob(mFileActivity);
 
                 /// immediate content synchronization

